runner/master_log.py

```diff
@@ -193,6 +191,23 @@ def as_context(self, test_id: str = ""):
 
         return context
 
+    def tokens_before_event(self, event: LogEvent) -> int:
+        event_idx = self.event_idx(event)
+
+        tokens = 0
+        for evt in self.log[:event_idx]:
+            if evt.type in [EventType.SEND_MESSAGE, EventType.SEND_FILL, EventType.RESPONSE_MESSAGE, EventType.RESPONSE_FILL]:
+                tokens += token_len(evt.data["message"])
+
+        return tokens
```

Comment:
I admit that this works, but maybe a bit overkill? What about storing the token count in the test start event?